web/src/main/java/org/springframework/security/web/server/authentication/SwitchUserWebFilter.java

```diff
@@ -0,0 +1,366 @@
+		return new AuthenticationCredentialsNotFoundException(
+				"No target user for the given username"
+		);
+	}
+
+	private static class SwitchUserAuthenticationException extends RuntimeException {
+		SwitchUserAuthenticationException(AuthenticationException exception) {
+			super(exception);
+		}
+	}
+
+	/**
+	 * Sets the repository for persisting the SecurityContext. Default is {@link WebSessionServerSecurityContextRepository}
+	 *
+	 * @param securityContextRepository the repository to use
+	 */
+	public void setSecurityContextRepository(
+			ServerSecurityContextRepository securityContextRepository) {
+		Assert.notNull(securityContextRepository, "securityContextRepository cannot be null");
+		this.securityContextRepository = securityContextRepository;
+	}
+
+	/**
+	 * Sets the userDetails object validator
+	 *
+	 * @param userDetailsChecker The <tt>userDetailsChecker</tt> which will be used to check
+	 *                           the data for the user that is being switched to.
+	 */
+	public void setUserDetailsChecker(UserDetailsChecker userDetailsChecker) {
+		this.userDetailsChecker = userDetailsChecker;
+	}
+
+	/**
+	 * Set the URL to respond to exit user processing. This is a shortcut for
+	 * * {@link #setExitUserMatcher(ServerWebExchangeMatcher)}
+	 *
+	 * @param exitUserUrl The exit user URL.
+	 */
+	public void setExitUserUrl(String exitUserUrl) {
+		Assert.isTrue(UrlUtils.isValidRedirectUrl(exitUserUrl),
+				"exitUserUrl cannot be empty and must be a valid redirect URL");
+		this.exitUserMatcher = createMatcher(exitUserUrl);
+	}
+
+	/**
+	 * Set the matcher to respond to exit user processing.
+	 *
+	 * @param exitUserMatcher The exit matcher to use
+	 */
+	public void setExitUserMatcher(ServerWebExchangeMatcher exitUserMatcher) {
+		Assert.notNull(exitUserMatcher, "exitUserMatcher cannot be null");
+		this.exitUserMatcher = exitUserMatcher;
+	}
+
+	/**
+	 * Set the URL to respond to switch user processing. This is a shortcut for
+	 * {@link #setSwitchUserMatcher(ServerWebExchangeMatcher)}
+	 *
+	 * @param switchUserUrl The switch user URL.
+	 */
+	public void setSwitchUserUrl(String switchUserUrl) {
+		Assert.isTrue(UrlUtils.isValidRedirectUrl(switchUserUrl),
+				"switchUserUrl cannot be empty and must be a valid redirect URL");
+		this.switchUserMatcher = createMatcher(switchUserUrl);
+	}
+
+	/**
+	 * Set the matcher to respond to switch user processing.
+	 *
+	 * @param switchUserMatcher The switch user matcher.
+	 */
+	public void setSwitchUserMatcher(ServerWebExchangeMatcher switchUserMatcher) {
+		Assert.notNull(switchUserMatcher, "switchUserMatcher cannot be null");
+		this.switchUserMatcher = switchUserMatcher;
+	}
+
+	/**
+	 * Allows the parameter containing the username to be customized.
+	 *
+	 * @param usernameParameter the parameter name. Defaults to {@code username}
+	 */
+	public void setUsernameParameter(String usernameParameter) {
```

Comment:
This is another method that I don't see as being a common use-case and I think we could leave out.

Reply:
Method has been removed.